doc/content/components/network-server.md

```diff
@@ -7,3 +7,54 @@ weight: 3
+
+## Application Downlink Queue Management and Linking
+
+Network Servers expose [AsNs]({{< ref "/reference/api/network_server#the-asns-service" >}}) service for application downlink management and linking. Typical clients of this service are [Application Servers]({{< ref "/components/application-server.md" >}}).
+
+Change of application downlink queue may trigger a downlink message.
+
+Once the link is established, Network Server will send application-specific uplink messages to the client via the link. There can be at most one active link per-appplication.
+
+In case link is not active, but Network Server has application-specific uplink messages to send, those messages will be queued and sent once the link is established.
+
+## Downlink Scheduling
+
+Network Server maintains internal downlink task queue. Each downlink task has an execution time associated with it, which tasks are sorted by in ascending order. Whenever a downlink task is ready to execute, it is executed as soon as possible.
+
+### Join-accept
+
+In case a pending session exists and join-accept is queued for the device, it is scheduled.
+
+### Data downlink
+
+In case a pending session does not exist or join-accept for it has already been sent, Network Server attempts to genererate and schedule data downlink in the active session.
+
+## Uplink Handling
+
+Network Servers expose [GsNs]({{< ref "/reference/api/network_server#the-gsns-service" >}}) service for uplink handling. Typical clients of this service are [Gateway Servers]({{< ref "/components/gateway-server.md" >}}).
+
+Network Servers process the uplinks received and handle accordingly. The first step is matching the uplink to a device. In case an uplink cannot be matched to a device stored in Network Server, it is dropped.
+
+### Join-request
+
+If join-request is received:
+1. Device is matched using the `DevEUI` and `JoinEUI` pair present in the join-request, which uniquely identifies the device.
+2. New `DevAddr` is assigned to the device and new MAC state is derived for the device.
+3. If [Join Server]({{< ref "/components/join-server.md" >}}) is present in the cluster, Network Server sends a join-request message to the cluster-local [Join Server]({{< ref "/components/join-server.md" >}}).
+4. If [Join Server]({{< ref "/components/join-server.md" >}}) is not present in the cluster or the device is not provisioned in the cluster-local [Join Server]({{< ref "/components/join-server.md" >}}), Network Server sends a join-request message to the [Join Server]({{< ref "/components/join-server.md" >}}) discovered via [interoperability configuration]({{< ref "/reference/interop-repository" >}}).
+5. If a [Join Server]({{< ref "/components/join-server.md" >}}) accepted the join-request, join-accept message may be enqueued for the device and application-specific uplink messages carrying relevant information about join-accept is sent to linked [Application Server]({{< ref "/components/application-server.md" >}}).
```

Comment:
I'm OK with the flow description, but imo this does not belong in the NS documentation. It would be a higher level description of how components interact on key LoRaWAN procedures